<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
   <TITLE>Sequery Information &amp; Documentation -- Use</TITLE>
   <META NAME="GENERATOR" CONTENT="Mozilla/3.0Gold (X11; I; SunOS 5.5.1 sun4m) [Netscape]">
</HEAD>
<BODY TEXT="#330000" BGCOLOR="#FFF8DC" LINK="#CC0000" VLINK="#DC143C" ALINK="#DC143C">

<P><A NAME="TOC"></A></P>

    <H1 ALIGN=CENTER>Using <i>Sequery</i></H1>

<CENTER><P><I>A tool to find sequence pattern matches in the Protein
	  Data Bank (PDB) database of protein structures.</I> </P></CENTER>

<hr size=5>

<OL>
<LI><A HREF="#Intro">Introduction</A></LI>

<LI><A HREF="#Install">Getting and Installing <I>Sequery</I></A></LI>

<LI><A HREF="#Library">Library Files Included with <I>Sequery</I></a></LI>

<LI><A HREF="#Running">Running <I>Sequery</I></A></LI>

<ul>
<li><A href="#Patterns">Sequence Query Patterns</a>
<li><A href="#Examples">Example Files</A>
</ul>

<LI><A HREF="#Output">Output Files Produced by <I>Sequery</I></A></LI>

<LI><A HREF="#Warnings">Bugs and Error Messages</A></LI>

<LI><A HREF="#Tools">Other Scripts and Tools Provided with <I>Sequery</I></A></LI>

<LI><A HREF="#Det">Algorithmic Details of <I>Sequery</I></A></LI>

<LI><A HREF="#Contact">Contact Information</A></LI>
</OL>

<hr size=5>

<H2><A NAME="Intro"></A>Introduction</H2>

<P><I>Sequery</i> is a tool to search the sequences of the protein
   structures in the <a href="http://www.pdb.bnl.gov">Protein Data
   Bank</a> (PDB) for a particular pattern of residues, which may
   include exact matches and acceptable substitutions based on a
   user-specified amino acid substitution matrix and/or a numerical
   threshold. <I>Sequery</i> was developed by Michael E. Pique,
   Michael A. Siani, Leslie A. Kuhn, Elizabeth D. Getzoff, and John
   A. Tainer, <A href="http://www.scripps.edu/mb">Department of
   Molecular Biology</a>, <A href="http://www.scripps.edu">The Scripps
   Research Institute</a> and is distributed here with permission of
   the authors. Development of <I>Sequery</i> was supported in part by
   NSF grant BIR9631436.</P>


<P>A useful complement to <I>Sequery</i> is <I>SSA</i>
   (Superpositional Structure Assignment), which automates the
   assignment of secondary structure to tetrapeptides identified by a
   <I>Sequery</i> search. For more information and software
   availability, see <A href="../ssa/doc.html"><I>SSA</i> Information and
   Documentation</a>.</P>

<P>For literature references related to <I>Sequery</I>, please see the
      section on <A HREF="#Det">Algorithmic Details</A>.</P>


<P><A HREF="#TOC">Back
to Table of Contents</A></P>

<hr size=5>

<H2><A NAME="Install"></A>Getting and Installing <I>Sequery</I></H2>

<P>Information on obtaining and installing <i>Sequery</i> can be found
      in <a href="installation.html">Installing <i>Sequery</i></a>.

<P><A HREF="#TOC">Back
to Table of Contents</A></P>

<hr size=5>

<H2><A NAME="Library"></A>Library Files Included with <I>Sequery</I></H2>
    
    Several library files are included in the <tt>sequery/lib</tt>
    directory for use with <i>Sequery</i>. The use of these files is
    explained later in <a href="#Running">Running <I>Sequery</I></a>.

    <ul>
      <LI><tt>pdbselectAug98.list</tt> -- The PDB Select
	list including PDB sequences with <25% pairwise identity from
	August 1998. Further information on the PDB Select list can be
	found at <A href="http://swift.embl-heidelberg.de/pdbsel/">PDB Select List
	  Site</a> at EMBL.
      <LI><tt>pdbselect98Aug.all.asc</tt> -- <I>Sequery</i> sequence
	file generated from all entries in the Aug 1998 25% threshold
	PDB Select list. (Includes both crystallographically-solved and
	NMR-solved protein structures.) In the distribution package,
	this is the default sequence file.
      <LI><TT>pdbselect98Augseq.xtalonly.asc</tt> -- <I>Sequery</i> sequence
	file generated from entries in the Aug 1998 25% threshold
	PDB Select list whose structures were solved via x-ray
	crystallography for studies in which one wishes to use
	restrict the search space to crystallographically solved
	structures.
      <LI><TT>pdbComplete.asc</tt> -- <i>Sequery</i> sequence file
	generated from the complete PDB database
      <LI><TT>sequery.defs</tt> -- Default <i>Sequery</i> sequence
	file read during execution. In the distribution package, this
	is a copy of <TT>pdbselect98Aug.all.asc</tt>
    </ul>

<P><A HREF="#TOC">Back
to Table of Contents</A></P>

<hr size=5>

<H2><A NAME="Running"></A>Running <I>Sequery</I></H2>

<P><I>Sequery</i> runs by searching for matching sequences (strings)
      in a precomputed ascii text file which contains sequences for
      each of the PDB files (or a specific subset of PDB
      files). <I>Sequery</i> is run using input supplied on stdin,
      either by redirection of a file containing the sequence patterns
      to match, see end of this section for an example, or by
      prompting the user for each pattern. It is common to run
      <I>Sequery</i> using tetrapeptide matching patterns since
      sequences of this length or shorter often have close matches in
      the PDB, but any length is possible. Execution is stopped via
      <tt>ctrl-D</tt>. Command line options are as follows:

<table border=0>

<tr>
<td valign=top><tt>-s&nbsp;SequenceFile</tt></td> <td valign=top>The
	  <tt>SequenceFile</tt> contains the listing of PDB codes and
	  corresponding amino acid sequences. If omitted,
	  <i>Sequery</i> defaults to searching
	  <tt>sequery/lib/pdbseq.asc</tt>. <I>Sequery</i> uses this
	  list to search for sequence pattern matches. This list can
	  be generated using either <tt>genpdbseq</tt> or
	  <tt>genpdbselectseq</tt>. Using <tt>genpdbselectseq</tt> is
	  the suggested method, including only sequences with less
	  than 25% identity, as this eliminates statistical bias
	  introduced by including structurally related proteins. For
	  more information, see the <A href="#Scripts">Additional
	  Scripts</A> section below. Note that any sequence database
	  formatted according to the example below may be
	  used.<P></td></tr>

      <tr><td colspan=2> Example portion of a sequence file (excerpt
	 from <tt>pdbselectAug98.all.asc</tt>):<PRE>
1hmc B    4    143 SEYCSHMIGSGHLQSLQRLIDSQMETSCQITFEFVDQEQLKDPVCYLKKA
                   FLLVQDIMEDTMRFRDNTPNAIAIVQLQELSLRLKSCFTKDYEEHDKACV
                   RTFYETPLQLLEKVKNVFNETKNLLDKDWNIFSKNCNNSFAEC
1myp A    1    104 CPEQDKYRTITGMCNNRRSPTLGASNRAFVRWLPAEYEDGFSLPYGWTPG
                   VKRNGFPVALARAVSNEIVRFPTDQLTPDQERSLMFMQWGQLLDHDLDFT
1grx _    1     84 MQTVIFGRSGC(13)YSVRAKDLAEKLSNERDDFQYQYVDIRAEGITKED
                   LQQKAGKPVETVPQIFVDQQHIGGYTDFAAWVKENLDA
                   PEPA
	  </pre>
</td></tr>
<tr><td></TD><td>
	  The fields are the PDB code, the chain identifier ( ``_''
	  indicates that there are no chain ID's in this structure),
	  the first residue number of the chain, the last residue
	  number of the chain, and the sequence of the chain. In some
	  structures, such as 1grx, the sequence field will contain
	  residue number. These indicate an instance of non-sequential
	  number in the sequence, e.g. due to the lack of diffractive
	  density for a mobile loop in the protein, and are used to
	  maintain correct sequence numbering for <I>Sequery</i>
	  output.<P>
</td>
</tr>

<tr>
<td valign=top><tt>-d&nbsp;DefinitionFile</tt></td> <td valign=top>The
	  <tt>DefinitionFile</tt> is a file containing acceptable
	  amino acid substitutions. If omitted, <i>Sequery</i>
	  defaults to using <tt>sequery/lib/sequery.defs</tt>. The
	  supplied substitution file with each line corresponding to a
	  set or equivalence class of substitutable amino acids,
	  <tt>sequery/lib/sequery.defs</tt>, was determined based on
	  the Dayhoff mutation data matrix, although any set of
	  substitutions could be provided in this format. When
	  entering the sequence pattern for a <I>Sequery</i>, an
	  upper-case charater indicates a search for an exact match
	  while a lower-case character indicates that all equivalent
	  residues from this file may be considered as substitutes
	  (e.g. ``A'' to match alanine only and ``a'' for all residues
	  equivalent to alanine). Further details can be found below
	  in <A href="#Patterns">Sequence Query Patterns</a>.</td>
</tr>

<tr>
<td valign=top><tt>-w&nbsp;WildcardFile</tt></td>
<td valign=top>The <tt>WildcardFile</tt> contains a listing of
	  user-defined acceptable amino acid substitutions (again,
	  with each line containing a set of amino acids that can
	  substitute for each other), e.g. from acceptable variation
	  observed in a sequence alignment or from mutagenesis
	  studies. When entering sequence patterns during
	  <I>Sequery</i> execution, the user enters the line number
	  within the <tt>WildcardFile</tt> corresponding to the
	  acceptable amino acids at that position. For example, based
	  on the example <tt>WildcardFile</tt>
	  (<tt>sequery/lib/wilddef.dat</tt>), entering a 2AAA would
	  find all patterns starting with tyrosine, phenylalanine, or
	  tryptophan (line 2 in the file), followed by 3
	  alanines.</td>
</tr>

<tr>
<td valign=top><tt>-o&nbsp;OutputFile</tt></td>
<td valign=top>The <tt>OutputFile</tt> is the file where the user
	  would like output to be placed. If omitted, output will be
	  written to <tt>sequery.match</tt>, overwriting any
	  previously existing <tt>sequery.match</tt>.

</table>

Other command line options include the following:

<table border=0>

<tr>
<td valign=top><tt>-x&nbsp;NumberOfContextResidues</tt></td> <td
valign=top>This is the number of residues printed (in lower-case) on
	  either side of the matched sequence pattern (in
	  upper-case). Default is 4.</td>
</tr>

<tr>
<td valign=top><tt>-v</tt></td>
<td vailgn=top>verbose mode -- More output (mostly for debugging
	  purposes)</td>
</tr>

<tr>
<td valign=top><tt>-q</tt></td>
<td valign=top>quiet mode -- Suppresses all output to the screen
	  except for error messages. Pattern matches will still be
	  output to the output file. </td>
</tr>

<tr>
<td valign=top><tt>-?</tt> or <tt>-h</tt></td>
<td valign=top>Gives version and help info</td>
</tr>

</table>
<P></P>
<I>Sequery</i> can be run in batch mode via the following (where
    <tt>search.patterns</tt> contains one line for each pattern to
    search):
<PRE>
      sequery -s lib/pdbseq.asc -d lib/sequery.defs \
         -w wilddef.dat -q -o search.matches < search.patterns
</pre>

<P><A HREF="#TOC">Back
to Table of Contents</A></P>

<hr size=5>

<H2><A NAME="Patterns"></A>Sequence Query Patterns</H2>

<P><I>Note: The following assumes the user is running with the default
	definition and wildcard files. </i></P>

<P>Sequence query patterns follow the regular expression rules, as
      documented in the UNIX ``ed'' man page. Following is a brief
      overview of pattern rules followed by several examples:</P>

    <UL>
      <B>Rules Summary</B>
      <LI>Upper-case letter -- match character (single-leter code for
        amino acid) exactly
      <LI>Lower-case letter -- match amino acid according to allowed
	substitutions in the definition file
      <LI><B>.</B> or X -- match any amino acid
      <LI>Digit -- match any of the amino acids in that line number of
	the wildcard file
      <LI>[Upper-case letters] -- match any of these amino acids
	(inside the brackets) in this position.
      <LI>[Lower-case letters] -- match any of these amino acids
        (inside the brackets), allowing for substitutions according to
        the definition file
      <LI>[^Upper-case letters] -- match any sequence not containing
	any of these amino acids at this
	position. (i.e. all other amino acids.)
      <LI>[^Lower-case letters] --match any sequence not containing
	any of these amino acids or their substitutions as
	allowed for in the definition file at this
	position. (i.e. match all other amino acids.)
      <LI>r\{m,n\} -- match m to n repetitions of the character r, 
        where m and n are digits. (Note that each digit must be
        preceded by a \ for <i>Sequery</i>, since digits without the \
        result in wildcard matching).  Example: 3 to 7 residues of any
        kind: .\{\3,\7\}; r can be upper-case (exact match) or
        lower-case (substituted match).<P></P>
      <B>Query Examples</B>
      <LI>ACYE -- search for Ala-Cys-Tyr-Glu
      <LI>acye -- expands to [A][C][Y][DQEBZ] according to the
	definitions file -- search for Ala-Cys-Tyr-(Asp or Gln or Glu
	or Asx or Glx)
      <LI>[AC]YEA -- search for (Ala or Cys)-Tyr-Glu-Ala
      <LI>[^ACDE]YEA -- search for (not(Ala or Cys or Asp or
	Glu))-Tyr-Glu-Ala. This is effectively a search for Tyr-Glu-Ala
	not preceeded by Ala, Cys, Asp, or Glu.
      <LI>AC.E or ACXE -- search for Ala-Cys-any single residue-Glu
      <LI>AC2E -- expands to AC[YFW]E according to line 2 of the
	wildcard file.
      <LI>A\{\3,\7\} -- expands to search for AAA, AAAA, AAAAA,
	AAAAAA, or AAAAAAA. (3 to 7 Ala's in a row)
    </ul>

<P><A HREF="#TOC">Back
to Table of Contents</A></P>

<hr size=5>

<H2><A NAME="Examples"></A>Example Files</H2>

There are example input, output, and wildcard files included in the
    <TT>sequery/examples</tt> directory. The example output was
    generated with the following command (run in the examples directory):
<PRE>
../bin/sequery -d ../lib/sequery.defs -s ../lib/pdbseq.asc \
     -w example.wilddef.dat -o example.matches < example.patterns
</pre>

<P><A HREF="#TOC">Back
to Table of Contents</A></P>

<hr size=5>

<H2><A NAME="Output"></A>Output Files Generated by <I>Sequery</I></H2>

<P><I>Sequery</i> will output the results of the pattern queries to the
    supplied output file (or to sequery.match if no output filename is
    given). Output will appear as follows:</P>
<PRE>
1cem _  153 to  156 -> aatdADEDiala matching ADED
1occ A   93 to   96 -> apdmAFPRmnnm matching 1234
</PRE>
Explanation:<BR>

<table border=0 width="80%">

<tr>
<td valign=top><B>Line 1</B></td>
<td width="80%" valign=top><i>PDB Code</i>: 1cem<br>
<I>Chain ID</i>: the underscore indicates there was no chain ID
	  specified in the PDB file. <br>
<I>Matching Residues</i>: 153-156<br>
<I>Matching &amp; Flanking Sequence</i>: <TT>aatdADEDiala</tt> (upper-case is
	  matched sequence, lower-case is flanking sequence)<BR>
<I>Query Pattern</I>: ADED
</td></tr>

<tr>
<td valign=top><B>Line 2</B></td>
<td valign=top><i>PDB Code</i>: 1occ<br>
<I>Chain ID</i>: A<br>
<I>Matching Residues</i>: 93-96<br>
<I>Matching &amp; Flanking Sequence</i>: <TT>apdmAFPRmnnm</tt> (upper-case is
	  matched sequence, lower-case is flanking sequence)<BR>
<I>Query Pattern</I>: one character each from lines 1,2,3, and 4 from
	  wildcard file
</td></tr>

</table>

<P><A HREF="#TOC">Back
to Table of Contents</A></P>

<hr size=5>

<H2><A NAME="Warnings"></A>Bugs and Error Messages</H2>


    <P>Most errors will occur due to improper query patterns. These
      errors will appear simply as non-run queries. The current
      version of <I>Sequery</i> shows unpredictable behavior with
      proteins with residues having negative residue numbers and will
      occasionally produce segmentation faults if sequence patterns
      would result in a very large number of matches. (In this case,
      break the query into two or more subqueries and combine the
      results.)

<P> <A HREF="#TOC">Back to Table of Contents</A></P>
 
<hr size=5>
 
<H2><A NAME="Tools"></A>Other Scripts and Tools Provided with
<I>Sequery</I></H2>
    Included in the <tt>sequery/share</tt> directory are a few
    additional tools that may be useful and are described below. These
    scripts must be modified upon installation to point to the correct
    directories. See <A href="installation.html#Install">Installing
      <I>Sequery</i></a> for more information.
<UL>

      <LI>addname -- adds protein name from the PDB file to the end of
	a <I>Sequery</i> output line. Output is to stdout. This is
	useful for noting functional patterns in the proteins that
	match.<P>
	Syntax:<br>
	<TT>addname SequeryOutputFile [columns]</tt><BR>
	  &nbsp;&nbsp;&nbsp;<tt>SequeryOutputFile</tt>: file generated
	  by sequery<BR>
	  &nbsp;&nbsp;&nbsp;<tt>columns</tt>: total number of columns of each of line of
	  output (default 80). If not specified, output is truncated to 80
	  chars per line.<P>
      <LI>genpdbselectseq -- generates a sequence file which contains
	the amino acid sequences, along with PDB codes and chain IDs,
	which is used by <I>Sequery</i> to search for pattern
	matches, for each of the PDB files in the PDB Select list. 
	For more information about the PDB Select list of
	representative (<25% sequence identity) protein structures and
	the latest version, see the <a
	  href="http://swift.embl-heidelberg.de/pdbsel/">PDB Select List
	  Site</a> at EMBL. For more information regarding the
	sequence file, see above under <a href="#Running">Running
	  <i>Sequery</i></a>.<P>
	  Syntax:<BR>
	  <TT>genpdbselectseq  pdb-select-list-file
	    output-sequence-file</TT><P>

	  Explanation of use: The PDB Select list is a list of all
	  proteins/chains in the PDB whose sequence identity is less
	  than a certain percentage. (There are different lists for
	  different identity threshold levels.) Each chain in the list
	  represents a set of related chains. By using the
	  lowest-identity threshold (25%), any structural bias in
	  <I>Sequery</i> analysis is minimized. (This bias arises from
	  the fact that if a sequence query identifies sequences in a
	  series of related proteins whose structures are known, any
	  subsequent structural analysis will contain more bias
	  towards these related structures.)<P>

      <LI>genpdbseq -- generates a sequence file as a <I>Sequery</i>
	search database from a
	PDB file or series of PDB files. Output is directed to stdout.<P>
	  Syntax:<BR>
	  <TT>genpdbseq PDBFile > SequenceFile</tt><BR>
	  Examples:<BR>
	  <tt>genpdbseq 2sod.pdb > 2sod.ascseq</tt><BR>
	  <tt>genpdbseq *.pdb > pdb.ascseq</tt><P>

      <LI>minipdbextract -- generates a PDB formatted file for the
	residues in each line of <i>Sequery</i> output. It takes the
	start residue, end residue, and pdbcode from <tt>Sequery</tt>
	output, searches the <tt>$PDBHOME</tt> database for that
	protein, and extracts coordinate lines from the PDB files.<P>
	  Sytnax:<BR>
	<tt>minipdbextract SequeryOutputFile [x]</tt><BR>
	  &nbsp;&nbsp;&nbsp;<tt>SequeryOutputFile</tt>: file generated by
	  sequery<br>
	  &nbsp;&nbsp;&nbsp;<tt>x</tt> (optional): Number of flanking residues to
	  include (default is 0)<BR>

</ul>

<P><A HREF="#TOC">Back
to Table of Contents</A></P>

<hr size=5>

<H2><A NAME="Det"></A>Algorithmic Details of <I>Sequery</I></H2>

    <P><I>Sequery</i> was developed as a successor to <I>Searchwild</i>,
      which is described in</P>
    
    <P>Collawn JF, Kuhn LA, Liu LF, Tainer JA, Trowbridge IS<BR>
      <STRONG>Transplanted LDL and mannose-6-phosphate receptor
	internalization signals promote high-efficiency endocytosis of
	the transferrin receptor</strong><BR>
      <I>EMBO J.</i>, <B>10(11)</b> (Nov): 3247-3253 (1991)</P>
    
    <P>Collawn JF, Stangel M, Kuhn LA, Esekogwu V, Jing SQ, Trowbridge
      IS, Tainer JA<BR>
      <STRONG>Transferrin receptor internalization sequence YXRF
	implicates a tight turn as the structural recognition motif
	for endocytosis</strong><br>
      <I>Cell</I>, <B>63(5)</b> (Nov 30): 1061-1072 (1990)</P>

    <P>Other references related to the use of <I>Sequery</i> include the
      following:</P>

    <P>Craig L, Sanschagrin PC, Rozek A, Lackie S, Kuhn LA, Scott JK<BR>
      <A HREF="../abstracts/jmb_antibody.html">
	<STRONG>The Role of Structure in Antibody Cross-Reactivity Between
	  Peptides and Folded Proteins</STRONG></A><BR>
      <I>J. Mol. Biol.</i>, <B>281(1)</b> (Aug 7): 183-201 (1998)</P>

    <P>Chang CP, Lazar CS, Walsh BJ, Komuro M, Collawn JF, Kuhn LA,
      Tainer JA, Trowbridge IS, Farquhar MG, Rosenfeld MG<BR>
      <STRONG>Ligand-induced internalization of the epidermal growth
	factor receptor is mediated by multiple endocytic codes
	analogous to the tyrosine motif found in constitutively
	internalized receptors</strong><BR>
      <I>J. Biol. Chem.</i>, <B>268(26)</b> (Sep 15): 19312-19320
      (1993)</P>

<P><A HREF="#TOC">Back
to Table of Contents</A></P>

<hr size=5>

<H2><A NAME="Contact"></A>Contact Information</H2>

<P>Questions should be directed to Dr. Leslie Kuhn at:
<ADDRESS><A
	HREF="mailto:kuhn@agua.bch.msu.edu">kuhn@agua.bch.msu.edu</A></ADDRESS><P> 
or to Michael Pique at:
<address><a href="mailto:mp@scripps.edu">mp@scripps.edu</a></address>
</OL>

</BODY>
</HTML>
